superset/assets/javascripts/explore/components/ExploreViewContainer.jsx

```diff
@@ -157,27 +151,6 @@ class ExploreViewContainer extends React.Component {
     }
   }
 
-  addHistory({ isReplace = false, title }) {
```

Comment:
if you removed this function, when user playing with chart controls, and they refresh their page (for any reason, without saving slice), will they lost the current query state? not all the query parameters are saved in slice yet.

Reply:
They will. I discussed this with @mistercrunch a couple weeks back and at that point we were okay with this side effect for a more consistent redirect implementation.

Reply:
In airbnb many users didn't realize there is a share linked button. they still get used to share their work by directly copy-n-paste url from browser. i guess this change will be dangerous to them :)

Reply:
@graceguo-supercat would you like to revisit/or discuss that decision again?

Reply:
@ttannis can we just keep `addHistory` as untouched as possible?

Reply:
i feel it's safe to keep query state in browser history.
- we still need to keep url scheme backward compatible right? Even we display url in browser like /r/123, when user request with old scheme, we should handle it correctly.
- after query state added into browser history, there is no real long url request happened unless user refreshed browser or someone accessed the shared url. So i guess the rate of old url scheme is pretty low.